arbnode/espresso_caff_node.go

```diff
@@ -467,7 +467,7 @@ func (n *EspressoCaffNode) Start(ctx context.Context) error {
 		return fmt.Errorf("failed to start espresso streamer: %w", err)
 	}
 	// This is +1 because the current block is the block after the last processed block
-	currentBlockNum := n.executionEngine.Bc().CurrentBlock().Number.Uint64() + 1
+	currentBlockNum := n.executionEngine.Bc().CurrentBlock().Number.Uint64()
```

Comment:
I don't think we should change this. Although I think the name is confusing, it should be `nextBlockNumber` or something

Reply:
Changing this was how I fixed the caff node system test. It was fetching an incorrect message for the first message. This line would start the streamer on message position 1 as opposed to 0 to execute through the entire history of the chain. It was leading to a nonce too low error on the second block it the caff node would try to produce.

I think an equally valid solution would be to leave it like this to start at the next valid block in all cases but when `CurrentBlock().Number` is zero. In that case we should start the espresso streamer at message position zero. 

Perhaps my interpretation of the cause of the test failure was wrong, but this change made the test pass.